pkg/controllers/util.go

```diff
@@ -32,10 +31,11 @@ func (q *QueuingEventHandler) OnAdd(obj interface{}) {
 }
 
 func (q *QueuingEventHandler) OnUpdate(old, new interface{}) {
-	if reflect.DeepEqual(old, new) {
-		return
+	oldObj := old.(metav1.Object)
+	newObj := new.(metav1.Object)
```

Comment:
Why not add `metav1.Object` to the function signature instead of `interface{}` ?

Reply:
We use this as an AddFunc, which has the signature `func(interface{})`. Go cannot realise that `func(metav1.Object)` matches `func(interface{})`.